### PR TITLE
feat: wasm x11 hash

### PR DESCRIPTION
**Pull request by markin-io**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    b[i] = s.charCodeAt(i);
+    i++;
+  }
+  return b;
+};
+
+const bytes2Int32Buffer = (b) => {
+  if (!b) return [];
+  const len = b.length ? (((b.length - 1) >>> 2) + 1) : 0;
+  const buffer = new Array(len);
```
Comment (shumkov) [inline lib/utils.js:29]: Is it ok, that it responds with Array instead of Buffer?

File: lib/x11.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @return {digest}
+ */
+const createDigest = (bindings) => {
+  /**
+   * @typedef {digest}
+   * @param {(string|Buffer|TypedArray|Array)} rawInput
+   * @param {number} inputFormat
+   * @param {number} outputFormat
+   */
+  const digest = (rawInput, inputFormat, outputFormat) => {
```
Comment (shumkov) [inline lib/x11.js:21]: Shouldn't we follow the existing interface https://nodejs.org/docs/latest-v16.x/api/crypto.html#hashupdatedata-inputencoding ?

**Comment by thephez** [comment]
What do you guys think of dropping the dashevo part and just publishing it as wasm-x11-hash? It's not exactly a dash-unique thing.

**Comment by shumkov** [comment]
@thephez I would agree

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Guide https://web.dev/emscripting-a-c-library/
```
[the remark this answers is not in the document]
Comment (markin-io) [reply build.sh]: Done

File: lib/x11.js (hunk 2 above)
Comment (markin-io) [reply lib/x11.js]: I don't think there's a need for that right now. Made a simple `digest(input, encoding)` API that always returns a buffer. We will upgrade library as long as there's a demand for that

File: lib/x11.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+const OUTPUT_HASH_SIZE = 32;
+
+/**
+ * @param bindings
+ * @return {digest}
+ */
+const createDigest = (bindings) => {
+  /**
+   * @typedef {digest}
+   * @param {(string|Buffer|Uint8Array)} input
```
Comment (shumkov) [inline lib/x11.js:13]: You don't need parenthesis here

**Review by shumkov — APPROVED** [review]
LGTM 👍

**Pull request merged**